### PR TITLE
[sms][test-suite] Handle null/undefined recipients & extend tests

### LINKED ISSUE #10277: SMS.sendSMSAsync should warn or throw for passing invalid parameters
Moving here the following issue started in [Expo Forum](https://forums.expo.io/t/sms-sendsmsasync-should-warn-or-throw-for-passing-invalid-parameters/43061)

This comes from a use case I had to struggle and understand, which took me too long time of debugging and frustrations.
It would be great if `expo-sms` could throw an error or at least a warning in dev stage to warn for `undefined`/`null` values (or any other invalid values) being passed to `SMS.sendSMSAsync`. 
The **real problem** is when the sms modal is open to send the message - it immediately closing and the result of `SMS.sendSMSAsync` is `cancelled`, without any way to know why...
Also, there's no documentation to explain this kind of behaviour, which makes it even harder to track the issue.

See example code:
```javascript
async function sendTextMessage(contact, text) {
  try {
      // assume `contact.phoneNumber` is undefined,
      const { result } = await SMS.sendSMSAsync(contact.phoneNumber, text)
      // 'result' here is 'canceled' for each number
    }
    showMessagesSuccessfullySentMessage()
  } catch (e) {
    console.error(e)
    showSmsSendingProblemError()
  }
}
```
Having a code, not working as expected without errors or warnings is much harder to debug and understand. Hence I would suggest to add here some clue for developers (dev stage)

### PR DESCRIPTION
**Pull request by kkafar**
# Why

Closes [#10277](https://github.com/expo/expo/issues/10277)

# How

Added check for null/undefined address in `packages/expo-sms/src/SMS.ts`. 
Right now, if any of given addresses is null/undefined TypeError is thrown. 

Additionally test-suite and ncl were extended to check this behaviour. 

# Test Plan

On Android device:

1. test-suite
2. ncl 

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).

### CODE REVIEW

**Comment by kkafar** [comment]
@lukmccall request for review

File: apps/test-suite/tests/SMS.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
   loadAttachmentsAsync,
   cleanupAttachmentsAsync,
   testSMSComposeWithSingleImageAttachment,
   testSMSComposeWithAudioAttachment,
   testSMSComposeWithTwoImageAttachments,
+  testSMSComposeWithNullRecipient,
+  testSMSComposeWithUndefinedRecipient,
 } from './SMSCommon';
 
+
```
Comment (lukmccall) [inline apps/test-suite/tests/SMS.js:16]:
```suggestion
```

**Review by lukmccall — APPROVED** [review]
LGMT 🚀🚀🚀

**Comment by lukmccall** [comment]
You need to rebuild js. See https://github.com/expo/expo/runs/3102472708

**Pull request merged**